newlib/libc/machine/riscv/memset.S

```diff
@@ -9,105 +9,275 @@
    http://www.opensource.org/licenses.
 */
 
+#include "sys/asm.h"
+
+
+#define BYTE_TBL_SZ		15
+#define WORD_TBL_SZ   64
+
+#if __riscv_zilsd || __riscv_zclsd
+#define SZREG         8
+#define	REG_S         sd
+#define PTRLOG        3
+#endif
+
+
 .text
 .global memset
 .type	memset, @function
+
+// void *memset(void *s, int c, size_t n);
+
 memset:
 #if defined(PREFER_SIZE_OVER_SPEED) || defined(__OPTIMIZE_SIZE__)
-  mv t1, a0
-  beqz a2, 2f
+  mv    t1, a0
+  beqz  a2, 2f
 
 1:
-  sb a1, 0(t1)
+  sb    a1, 0(t1)
   add   a2, a2, -1
   add   t1, t1, 1
-  bnez a2, 1b
+  bnez  a2, 1b
 
 2:
   ret
 
 #else
-  li t1, 15
-  move a4, a0
-  bleu a2, t1, .Ltiny
-  and a5, a4, 15
-  bnez a5, .Lmisaligned
+  mv		a3, a0
+	li		t6, BYTE_TBL_SZ
```

Comment:
make sure to not mix space and tab indention like this. It is usually a good idea to configure your editor to show the tabs and spaces explicity so you don't overlook these